src/parser/SparqlParser.cpp

```diff
@@ -985,3 +979,13 @@ ParsedQuery::Alias SparqlParser::parseAliasWithAntlr(
   _lexer.reset(std::move(resultOfParseAndRemainingText._remainingText));
   return std::move(resultOfParseAndRemainingText._resultOfParse);
 }
+
+// ________________________________________________________________________
+GraphPatternOperation::Bind SparqlParser::parseBindWithAntlr(
+    const ParsedQuery& parsedQuery) {
+  auto str = _lexer.getUnconsumedInput();
+  auto resultOfParseAndRemainingText =
+      sparqlParserHelpers::parseBind(str, getPrefixMap(parsedQuery));
+  _lexer.reset(std::move(resultOfParseAndRemainingText._remainingText));
+  return std::move(resultOfParseAndRemainingText._resultOfParse);
+}
```

Comment:
This function, and the two functions above look exactly the same, except for the name of the `sparqlParserHelpers::parseXXX` function. You can unify them to an abstraction that can be called like that:

```
// old: auto bind = parseBindWithAntlr(*query);
auto bind = parseWithAntlr<sparqlParserHelpers::parseBind>(*query);
```

If you need help in doing so using template magic, I can help you, just contact me.